lib/SPIRV/OCLUtil.cpp

```diff
@@ -805,7 +805,7 @@ unsigned encodeVecTypeHint(Type *Ty) {
       llvm_unreachable("invalid integer type");
     }
   }
-  if (VectorType *VecTy = dyn_cast<VectorType>(Ty)) {
+  if (FixedVectorType *VecTy = dyn_cast<FixedVectorType>(Ty)) {
```

Comment:
```suggestion
  if (auto *VecTy = dyn_cast<FixedVectorType>(Ty)) {
```

Type of `VecTy` is obvious here - good spot for using `auto`